[scroll-animations] JS override for animation-range should be per property

nmoucht opened the pull request:
#### c16514b883b241c4b56e47dee2837cd982a94354
<pre>
[scroll-animations] JS override for animation-range should be per property
<a href="https://bugs.webkit.org/show_bug.cgi?id=285883">https://bugs.webkit.org/show_bug.cgi?id=285883</a>
<a href="https://rdar.apple.com/142841187">rdar://142841187</a>

Reviewed by Antoine Quint.

Setting animation-range via JS should override only on the property set (ie setting animation-range-start
via JS API should only override animation-range-start and not animation-range-end as well).

* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/animation-range-ignored-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/animation-range-ignored.html:
* Source/WebCore/animation/CSSAnimation.cpp:
(WebCore::CSSAnimation::syncPropertiesWithBackingAnimation):
(WebCore::CSSAnimation::setBindingsRangeStart):
(WebCore::CSSAnimation::setBindingsRangeEnd):
* Source/WebCore/animation/CSSAnimation.h:
* Source/WebCore/animation/ViewTimeline.cpp:
(WebCore::ViewTimeline::intervalForAttachmentRange const):
* Source/WebCore/animation/WebAnimation.cpp:
(WebCore::WebAnimation::setRangeStart):
(WebCore::WebAnimation::setRangeEnd):
(WebCore::WebAnimation::setRange): Deleted.
* Source/WebCore/animation/WebAnimation.h:

Canonical link: <a href="https://commits.webkit.org/288973@main">https://commits.webkit.org/288973@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/77e064307d235271d171609ad93f8badced1424d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84892 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/4617 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/39282 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/90036 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35945 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/86977 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/4707 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/12594 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/66065 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23880 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87937 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/3595 "Found 1 new test failure: fast/forms/ios/file-upload-panel.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/77152 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46335 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/3471 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31379 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/35018 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/74289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/32186 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/91409 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/12231 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/8935 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/74546 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/12461 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72964 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/73670 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/18056 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16509 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/4266 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13233 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/12183 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/17634 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/12018 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/15512 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/13763 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

graouts commented on Source/WebCore/animation/WebAnimation.cpp:
```diff
@@ -1907,12 +1907,22 @@ void WebAnimation::setBindingsRangeEnd(TimelineRangeValue&& rangeEndValue)
 }
 
-void WebAnimation::setRange(TimelineRange range)
+void WebAnimation::setRangeStart(SingleTimelineRange range)
```
We should rename `range` to `rangeStart` here.

nmoucht replied:
Fixed.

graouts commented on Source/WebCore/animation/WebAnimation.cpp:
```diff
@@ -1907,12 +1907,22 @@ void WebAnimation::setBindingsRangeEnd(TimelineRangeValue&& rangeEndValue)
+        effect->animationRangeDidChange();
+}
+
+void WebAnimation::setRangeEnd(SingleTimelineRange range)
```
The name `rangeEnd` would be better for the parameter.

nmoucht replied:
Fixed.

nmoucht committed:
[scroll-animations] JS override for animation-range should be per property
https://bugs.webkit.org/show_bug.cgi?id=285883
rdar://142841187

Reviewed by Antoine Quint.

Setting animation-range via JS should override only on the property set (ie setting animation-range-start
via JS API should only override animation-range-start and not animation-range-end as well).

* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/animation-range-ignored-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/scroll-animations/css/animation-range-ignored.html:
* Source/WebCore/animation/CSSAnimation.cpp:
(WebCore::CSSAnimation::syncPropertiesWithBackingAnimation):
(WebCore::CSSAnimation::setBindingsRangeStart):
(WebCore::CSSAnimation::setBindingsRangeEnd):
* Source/WebCore/animation/CSSAnimation.h:
* Source/WebCore/animation/ViewTimeline.cpp:
(WebCore::ViewTimeline::intervalForAttachmentRange const):
* Source/WebCore/animation/WebAnimation.cpp:
(WebCore::WebAnimation::setRangeStart):
(WebCore::WebAnimation::setRangeEnd):
(WebCore::WebAnimation::setRange): Deleted.
* Source/WebCore/animation/WebAnimation.h:

Canonical link: https://commits.webkit.org/288973@main
